satpy/readers/utils.py

```diff
@@ -311,3 +312,33 @@ def apply_rad_correction(data, slope, offset):
     """Apply GSICS-like correction factors to radiance data."""
     data = (data - offset) / slope
     return data
+
+
+def apply_earthsun_distance_correction(reflectance, utc_date=None):
+    """Correct reflectance data to account for changing Earth-Sun distance."""
+    from pyorbital.astronomy import sun_earth_distance_correction
+    if utc_date is None:
+        try:
+            utc_date = reflectance.attrs['start_time']
+        except AttributeError:
+            try:
+                utc_date = reflectance.attrs['scheduled_time']
+            except AttributeError:
+                raise
+    se_dist = sun_earth_distance_correction(utc_date)
```

Comment:
Can you please rename `se_dist` to `se_dist_corr`? I was a bit confused because I thought it was the sun earth distance.

Reply:
It is the Sun-Earth distance :) The function it's calling (in pyorbital) should be renamed, as it is not a correction. I'll raise an issue for this.